app-backend/db/src/test/scala/com/azavea/rf/database/DatasourceDaoSpec.scala

```diff
@@ -201,35 +200,34 @@ class DatasourceDaoSpec
         (userCreate: User.Create,
          ownerCreate: User.Create,
          orgCreate: Organization.Create,
+         platform: Platform,
          dsCreate: Datasource.Create,
          project: Project.Create,
          upload: Upload.Create) => {
           @SuppressWarnings(Array("TraversableHead"))
           val isDsDeletableIO = for {
-            orgUserProject <- insertUserOrgProject(userCreate,
-                                                   orgCreate,
-                                                   project)
-            (dbOrg, dbUser, dbProject) = orgUserProject
+            orgUserPlatProject <- insertUserOrgPlatProject(userCreate,
+                                                           orgCreate,
+                                                           platform,
+                                                           project)
+            (dbUser, dbOrg, dbPlatform, dbProject) = orgUserPlatProject
             datasource <- fixupDatasource(dsCreate, dbUser)
-            statuses = List(UploadStatus.Created,
-                            UploadStatus.Uploading,
-                            UploadStatus.Uploaded,
-                            UploadStatus.Queued,
-                            UploadStatus.Processing)
-            status = Random.shuffle(statuses).head
```

Comment:
Since the arbitrary instance already randomly chooses a status, we don't need to add extra randomness here, and can just test the deletable condition against the status we got from the first die roll.